### PR TITLE
Salto-4054: unable to deploy statistical accounts in netsuite

### PR DESCRIPTION
**Pull request by Shulik95**
_fix bug preventing deployments of statistical accounts_

---

_None_

---
_Release Notes_: 

---
_User Notifications_: 
_id field will be added in the unitsType, mainAddress and state fields_

### CODE REVIEW

**Review by omrilit — COMMENTED** [review]
@dantal4 please review this one.
My main concern is the impact on existing WSs. Does it have footpront outside of the Types folder?

**Comment by Shulik95** [comment]
> @dantal4 please review this one. My main concern is the impact on existing WSs. Does it have footprint outside of the Types folder?
@omrilit It will affect existing WS since the id field will be added in some instance but I've implemented noise suppression for these changes - https://github.com/salto-io/salto_private/pull/5678

**Review by Shulik95 — COMMENTED** [review]
@omrilit It will affect existing WS but I've also opened a pr for noise suppression for these changes - https://github.com/salto-io/salto_private/pull/5678

**Comment by omrilit** [comment]
> @omrilit It will affect existing WS but I've also opened a pr for noise suppression for these changes - [salto-io/salto_private#5678](https://github.com/salto-io/salto_private/pull/5678)

We use the noise suppression to omit changes from the monitoring notifications, to avoid alerting users on noisy changes.
The thing I am concerned about is the Audit users, which will have to commit the affected files to Git, and we should always strive to notify them in advance so they could push the uncommitted files before we deploy the new version and then fetch and push only the noisy changes in a dedicated commit.
For example you can see https://salto-io.atlassian.net/browse/PROD-76.
I think that it would be great if we could have an example of the exact changes the users will experience so we could share it with them or with CS in advance (screenshots / a commit in a git repo with the changes in one of our accounts).

File: packages/netsuite-adapter/src/filters/data_instances_internal_id.ts [hunk 1]
```diff
@@ -39,10 +39,8 @@ const hasInternalIdHiddenField = (type: unknown): boolean =>
   && type.fields[INTERNAL_ID]
   && type.fields[INTERNAL_ID].annotations[CORE_ANNOTATIONS.HIDDEN_VALUE]
 
-const shouldUseIdField = (fieldType: TypeElement | undefined, path: ElemID): boolean =>
-  fieldType?.elemID.name === RECORD_REF || (
-    isCustomFieldName(path.name) && hasInternalIdHiddenField(fieldType)
-  )
+const shouldUseIdField = (fieldType: TypeElement | undefined): boolean =>
+  fieldType?.elemID.name === RECORD_REF || hasInternalIdHiddenField(fieldType)
```
Comment (dantal4) [inline packages/netsuite-adapter/src/filters/data_instances_internal_id.ts:43]: Worth checking if `netsuite.recordRef` has a hidden internal id field. If it has, we can just use `hasInternalIdHiddenField(fieldType)`

**Review by dantal4 — APPROVED** [review]
Great work!
Please add a test for the case that this PR now supports.

**Pull request merged**